test/jdk/java/lang/Class/getEnclosingClass/EnclosingClassTest.java

```diff
@@ -59,71 +48,172 @@
+
+    @BeforeClass
+    public void createEnclosingClasses() throws Throwable {
+        Path enclosingPath = Paths.get(ENCLOSING_CLASS_SRC);
+        Path pkg1Dir = Paths.get(SRC_DIR + "/pkg1");
+        Path pkg2Dir = Paths.get(SRC_DIR + "/pkg1/pkg2");
+        Path pkg1File = Paths.get(SRC_DIR + "/pkg1/EnclosingClass.java");
+        Path pkg2File = Paths.get(SRC_DIR + "/pkg1/pkg2/EnclosingClass.java");
+        try {
+            Files.deleteIfExists(pkg2File);
+            Files.deleteIfExists(pkg2Dir);
+            Files.deleteIfExists(pkg1File);
+            Files.deleteIfExists(pkg1Dir);
+            Files.createDirectories(pkg2Dir);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+        createAndWriteEnclosingClasses(enclosingPath, pkg1File, "pkg1");
+        createAndWriteEnclosingClasses(enclosingPath, pkg2File, "pkg1.pkg2");
+
+        String javacPath = JDKToolFinder.getJDKTool("javac");
+        OutputAnalyzer outputAnalyzer = ProcessTools.executeCommand(javacPath, "-d", System.getProperty("test.classes", "."),
+                SRC_DIR + "/EnclosingClass.java", SRC_DIR + "/pkg1/EnclosingClass.java", SRC_DIR + "/pkg1/pkg2/EnclosingClass.java");
+
+        outputAnalyzer.shouldHaveExitValue(0);
+    }
+
+    @Test
+    public void testEnclosingClasses() throws Throwable {
+        test(Class.forName("EnclosingClass").getDeclaredConstructor().newInstance());
+    }
+
+    @Test
+    public void testEnclosingClassesInPackage() throws Throwable {
+        test(Class.forName("pkg1.EnclosingClass").getDeclaredConstructor().newInstance());
+    }
+
+    @Test
+    public void testEnclosingClassesInNestedPackage() throws Throwable {
+        test(Class.forName("pkg1.pkg2.EnclosingClass").getDeclaredConstructor().newInstance());
+    }
+
+    @AfterClass
+    public void deleteEnclosingClasses() {
+        Path pkg1Dir = Paths.get(SRC_DIR + "/pkg1");
+        Path pkg2Dir = Paths.get(SRC_DIR + "/pkg1/pkg2");
+        Path pkg1File = Paths.get(SRC_DIR + "/pkg1/EnclosingClass.java");
+        Path pkg2File = Paths.get(SRC_DIR + "/pkg1/pkg2/EnclosingClass.java");
+        try {
+            Files.deleteIfExists(pkg2File);
+            Files.deleteIfExists(pkg2Dir);
+            Files.deleteIfExists(pkg1File);
+            Files.deleteIfExists(pkg1Dir);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    private void createAndWriteEnclosingClasses(final Path source, final Path target, final String packagePath) {
+        try (BufferedReader br = new BufferedReader(new FileReader(source.toFile()));
+        PrintWriter bw = new PrintWriter(new FileWriter(target.toFile()))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                if (line.contains("canonical=\"EnclosingClass")) {
+                    line = line.replaceAll("canonical=\"EnclosingClass", "canonical=\"" + packagePath + ".EnclosingClass");
```

Comment:
suggestion: declare `var className = packageName + ".EnclosingClass";`  and replace those occurrance.

Reply:
Thanks. Implemented the comment in next patch.